addon/components/polaris-choice-list.js

```diff
@@ -104,6 +121,22 @@ export default Component.extend({
    */
   onChange() {},
 
+  /**
+   * @private
+   */
+  hasError: bool('error'),
+
+  /**
+   * @private
+   */
+  finalNameError: computed('finalName', 'error', function() {
+    if (isNone(this.get('error'))) {
+      return null;
+    }
+
+    return `${this.get('finalName')}Error`;
```

Comment:
We have an existing [util](https://github.com/smile-io/ember-polaris/tree/master/addon/utils/id.js#L16) for generating this, maybe worth using that here? 🤷‍♂️